### PR TITLE
Stylus Dictionary Support

### PR DESCRIPTION
**Pull request by rachel-bousfield**
Adds an extra byte to the Stylus EOF header for selecting the custom dictionary

### CODE REVIEW

File: core/state/statedb_arbitrum.go [hunk 1]
```diff
@@ -39,31 +39,36 @@ var (
 	stylusEOFMagic       = byte(0xEF)
 	stylusEOFMagicSuffix = byte(0xF0)
 	stylusEOFVersion     = byte(0x00)
+	stylusEOFDictionary  = byte(0x00) // varies with dictionary selection
```
Comment (mikebenfield) [inline core/state/statedb_arbitrum.go]: I may be missing something, but this new variable is unused.

File: core/state/statedb_arbitrum.go [hunk 2]
```diff
@@ -39,31 +39,36 @@ var (
...
+	if len(b) < len(StylusDiscriminant)+1 {
 		return false
 	}
-	return bytes.Equal(b[:3], StylusPrefix)
+	return bytes.Equal(b[:3], StylusDiscriminant)
 }
 
-// StripStylusPrefix if the specified input is a stylus program.
-func StripStylusPrefix(b []byte) ([]byte, error) {
+// checks if the specified input is a stylus program.
```
Comment (mikebenfield) [inline core/state/statedb_arbitrum.go]: This comment seems to be wrong or incomplete. Maybe "If the input is a stylus program, return it with its prefix stripped as well as its version number." (Or dictionary number, or whatever.)

**Review by mikebenfield — COMMENTED** [review]
Made a couple remarks.

File: core/state/statedb_arbitrum.go (hunk 1 above)
Comment (rachel-bousfield) [reply core/state/statedb_arbitrum.go]: Changed to just be a comment

**Pull request merged**